zerver/openapi/zulip.yaml

```diff
@@ -5006,12 +5006,15 @@ paths:
         clients (e.g. the web, desktop, mobile, and terminal clients) as well
         as bots, API clients, backup scripts, etc.
 
-        By specifying a [narrow filter](/api/construct-narrow), you can use
-        this endpoint to fetch the messages matching any search query that is
-        supported by Zulip's powerful full-text search backend.
+        By specifying a [narrow filter](/api/get-messages#parameter-narrow),
+        you can use this endpoint to fetch the messages matching any search
+        query that is supported by Zulip's powerful full-text search backend.
 
         When a narrow is not specified, it can be used to fetch a user's
-        message history. (We recommend paginating to 1000 messages at a time.)
+        message history, which means that the results will be from streams
+        that the user is subscribed to and from private/group messages in
+        which the user is a participant. Note that bots are not automatically
+        subscribed to the same streams as the bot's creator.
```

Comment:
Do you think something like "Note that a user's personal message history does not contain messages sent to streams before they subscribe, and newly created bot users usually are not be subscribed to any streams." would be clearer here? And then perhaps we can swap the order of the paragraphs so that the "By specifying a narrow filter" paragraph comes just after might feel more natural. 

(I think you can via the API create a bot with initial stream subscriptions, and as noted elsewhere, I'm thinking we should probably add UI for that too).